src/components/newOrderForm.jsx

```diff
@@ -34,9 +34,22 @@ const NewOrderForm = React.createClass({
         });
     },
 
-    validateHour (hour) {
-        const pattern = /[0-9]{1,2}\:[0-9]{2}/;
-        return pattern.test(hour);
+    validateHour (hourInput) {
+        const pattern = /([0-9]{1,2})\:([0-9]{2})/;
+        if (!pattern.test(hourInput)) {
+            return false;
+        }
+
+        const matches = pattern.exec(hourInput);
```

Comment:
you can use [array destructuring](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Pulling_values_from_a_regular_expression_match) (es6) - there's a regex example :D